src/main/cpp/util/file_platform.h

```diff
@@ -31,10 +31,11 @@ class IFileMtime {
  public:
   virtual ~IFileMtime() {}
 
-  // Queries the mtime of `path` to see whether it's in the distant future.
-  // Returns true if querying succeeded and stores the result in `result`.
-  // Returns false if querying failed.
-  virtual bool GetIfInDistantFuture(const std::string &path, bool *result) = 0;
+  // Checks if `path` is a valid embedded binary, i.e. it is either an
+  // embedded binary that Bazel extracted from a previous run and it hasn't
+  // been tampered with (its mtime is unchanged), or it's a directory.
+  // Returns false otherwise, or if querying the information failed.
+  virtual bool IsValidEmbeddedBinary(const std::string &path) = 0;
```

Comment:
Ok, I hear you on not wanting to move this to blaze_util_platform. Maybe you can make this name more general then, though? IsUntamperedFile() keeps it more neutral.

Reply:
I like that idea! Done.

Reply:
I also added a TODO with my own name to move the function.